app/2.0/docs/devguide/dom-template.md

```diff
@@ -83,4 +244,140 @@ use the standard DOM `querySelector`  method.
 
 
 
+## Remove empty text nodes {#strip-whitespace}
+
+
+Add the `strip-whitespace` boolean attribute to a template to remove
+any empty text nodes from the template's contents. This can result in a
+minor performance improvement.
+
+With empty text nodes:
+
+
+```html
+<dom-module id="has-whitespace">
+  <template> <div>A</div> <div>B</div> </template>
+  <script>
+    class HasWhitespace extends Polymer.Element {
+      static get is() { return  'has-whitespace' }
+      ready() {
+        super.ready();
+        console.log(this.shadowRoot.childNodes.length); // 5
+      }
+    }
+    customElements.define(HasWhitespace.is, HasWhitespace);
+  </script>
+</dom-module>
+```
+
+
+Without empty text nodes:
+
+
+```html
+<dom-module id="no-whitespace">
+  <template strip-whitespace>
+    <div>A</div>
+    <div>B</div>
+  </template>
+  <script>
+    class NoWhitespace extends Polymer.Element {
+      static get is() { return  'no-whitespace' }
+      ready() {
+        super.ready();
+        console.log(this.shadowRoot.childNodes.length); // 2
+      }
+    }
+    customElements.define(NoWhitespace.is, NoWhitespace);
+  </script>
+</dom-module>
+```
+
+## Preserve template contents
+
+Polymer performs one-time processing on your DOM template, such as parsing and removing binding
+annotations and caching  and removing nested template contents for better performance. This processing
+removes the template's original contents (the `content` property will be undefined). If you want
+to access the contents of a nested template, you can add the `preserve-content` attribute to the
+template.
+
+Preserving the contents of a nested template means it **won't have any Polymer features like
+data bindings or declarative event listeners.** Only use this when you want to manipulate the
+template yourself, and you don't want Polymer to touch it.
+
+This is a fairly rare use case.
+
+```html
+<dom-module id="custom-template">
+  <template>
+    <template id="special-template" preserve-content>
+      <div>I am very special.</div>
+    </template>
+  </template>
+  <script>
+    class CustomTemplate extends Polymer.Element {
+
+      static get is() { return  'custom-template' }
+
+      ready() {
+        super.ready();
+        // retrieve the nested template
+        let template = this.shadowRoot.querySelector('#special-template');
+
+        //
+        for (let i=0; i<10; i++) {
+          this.shadowRoot.appendChild(document.importNode(template.content, true));
+        }
+      }
+    }
+
+    customElements.define(CustomTemplate.is, CustomTemplate);
+  </script>
+</dom-module>
+```
+
+## Customize DOM initialization
+
+There are several points where you can customize how Polymer initializes your element's DOM. You
+can customize how the shadow root is created by creating it yourself. And you can override the
+`_attachDom` method to change how the the DOM tree is added to your element: for example, to
+stamp into light DOM instead of shadow DOM.
```

Comment:
Perhaps should note that when stamping into light DOM, styling those elements becomes a problem that the user has to solve, e.g. by placing a stylesheet in the body, etc.